platform/security/sts/security-sts-ldapclaimshandler/src/main/java/ddf/security/sts/claimsHandler/RoleClaimsHandler.java

```diff
@@ -240,41 +241,49 @@ public ProcessedClaimCollection retrieveClaimValues(
 
         String membershipValue = user;
 
-        AndFilter filter;
-        ConnectionEntryReader entryReader;
-        if (!membershipUserAttribute.equals(loginUserAttribute)) {
-          String baseDN = AttributeMapLoader.getBaseDN(principal, userBaseDn, overrideCertDn);
-          filter = new AndFilter();
-          filter.and(new EqualsFilter(this.getLoginUserAttribute(), user));
-          entryReader =
-              connection.search(
-                  baseDN, SearchScope.WHOLE_SUBTREE, filter.toString(), membershipUserAttribute);
-          while (entryReader.hasNext()) {
-            if (entryReader.isEntry()) {
-              SearchResultEntry entry = entryReader.readEntry();
-
+        String baseDN = AttributeMapLoader.getBaseDN(principal, userBaseDn, overrideCertDn);
+        AndFilter filter = new AndFilter();
+        filter.and(new EqualsFilter(this.getLoginUserAttribute(), user));
+        ConnectionEntryReader entryReader =
+            connection.search(
+                baseDN, SearchScope.WHOLE_SUBTREE, filter.toString(), membershipUserAttribute);
+        String userDN = loginUserAttribute + "=" + user + "," + baseDN;
```

Comment:
✏️ `String.format` is is  more readable in my opinion

`String userDN = String.format("%s = %s,%s", loginUserAttribute, user, baseDN);`